packages/client/hmi-client/src/workflow/ops/decapodes/tera-decapodes-drilldown.vue

```diff
@@ -0,0 +1,209 @@
+<!-- FIXME: decapodes with LLM isnt entirely functional, this is more of a placeholder at the moment -->
+<template>
+	<tera-drilldown :title="node.displayName" @on-close-clicked="emit('close')">
+		<template #header-actions>
+			<tera-output-dropdown
+				@click.stop
+				:output="selectedOutputId"
+				is-selectable
+				:options="outputs"
+			/>
+		</template>
+		<main>
+			<tera-drilldown-section>
+				<template #footer>
+					<Button
+						icon="pi pi-play"
+						label="Run"
+						outlined
+						severity="secondary"
+						@click="runFromCode"
+						:loading="isRunningCode"
+					/>
+				</template>
+				<Suspense>
+					<tera-notebook-jupyter-input
+						:kernel-manager="kernelManager"
+						:defaultOptions="sampleAgentQuestions"
+						:context-language="contextLanguage"
+						@llm-output="(data: any) => appendCode(data, 'code')"
+					/>
+				</Suspense>
+				<v-ace-editor
+					v-model:value="codeText"
+					@init="initializeEditor"
+					lang="python"
+					theme="chrome"
+					style="flex-grow: 1; width: 100%"
+					class="ace-editor"
+				/>
+			</tera-drilldown-section>
+
+			<tera-drilldown-preview title="Output Preview">
+				<div>{{ notebookResponse }}</div>
+			</tera-drilldown-preview>
+		</main>
+	</tera-drilldown>
+</template>
+
+<script setup lang="ts">
+import { WorkflowNode } from '@/types/workflow';
+import TeraDrilldown from '@/components/drilldown/tera-drilldown.vue';
+import TeraDrilldownSection from '@/components/drilldown/tera-drilldown-section.vue';
+import TeraDrilldownPreview from '@/components/drilldown/tera-drilldown-preview.vue';
+import { KernelSessionManager } from '@/services/jupyter';
+import TeraNotebookJupyterInput from '@/components/llm/tera-notebook-jupyter-input.vue';
+import { VAceEditor } from 'vue3-ace-editor';
+import { VAceEditorInstance } from 'vue3-ace-editor/types';
+import { computed, onMounted, onUnmounted, ref, watch } from 'vue';
+import { logger } from '@/utils/logger';
+import Button from 'primevue/button';
+import TeraOutputDropdown from '@/components/drilldown/tera-output-dropdown.vue';
+import { cloneDeep, isEmpty } from 'lodash';
+import { getModel, getModelType } from '@/services/model';
+import { AMRSchemaNames } from '@/types/common';
+import { DecapodesOperationState } from './decapodes-operation';
+
+const props = defineProps<{
+	node: WorkflowNode<DecapodesOperationState>;
+}>();
+
+const emit = defineEmits(['close', 'update-state']);
+const kernelManager = new KernelSessionManager();
+let editor: VAceEditorInstance['_editor'] | null;
+
+const modelId = computed(() => props.node?.inputs?.[0]?.value?.[0]);
```

Comment:
`props.node` is not optional according to your declaration

Reply:
true, I've removed the optional for node.  I sort of like to computed here so that I can use it everywhere I need it in the notebook (which is only 2 places in this case 😄 ) rather than write out something like `modelId = props.node....` everywhere i need it but happy to change that if there an issue with it being computed